scripts/build.js

```diff
@@ -66,4 +66,55 @@ function getPackage() {
   }
 
   return undefined;
+}
+
+function loadTaskFunctions(tasks) {
+  return flatten(tasks)
+    .reduce((acc, taskName) => {
+      acc[taskName] = require('./tasks/' + taskName);
+      return acc;
+    }, {});
+}
+
+function flatten(list) {
+  return list.reduce((acc, tasks) => acc.concat(tasks));
+}
+
+// Filter disabled tasks if specified in the package.json.
+function removeDisabledTasks(disabledTasks = []) {
+  return tasks => tasks.filter(task => disabledTasks.indexOf(task) < 0);
+}
+
+function getTaskListAndTaskMap(process) {
+  const allTasks = [
```

Comment:
**ts/tslint/jest** can run in parallel, so **tslint/jest** could be started earlier than what you have, but if **ts** finishes before **jest** (likely for the biggest project in this repo) then **webpack** is waiting when it doesn't have to.

Consider a bit more complex system than this, where you indicate each task's dependency/ies.  

sass depends on copy
ts depends on sass
tslint depends on sass
jest depends on sass
webpack depends on ts

Use this dependency tree to set up promises stored in a dictionary and indexed by task name, and you can have a more efficient build.